_posts/2018-10-15-12-2-Use-SkipToken-for-Server-Side-Paging.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: "12.3 Use $skiptoken for server side paging "
+description: "WebAPI to use $skiptoken for server side paging"
+category: "12. Design"
+---
+# Use $skiptoken for server-driven paging
+
+### Background
+Loading large data can be slow. Services often rely on pagination to load the data incrementally to improve the response times and the user experience. Paging can be server-driven or client-driven:
+#### Client-driven paging
+In client-driven paging, the client decides how many records it wants to load and asks the server that many records. That is achieved by using $skip and $top tokens in conjunction. For instance, if a client needs to request 10 records from 71-80, it can send a similar request as below:
+
+`GET ~/Products/$skip=70&$top=10`
+#### Server-driven paging
+In server-driven paging, the client asks for a collection of entities and the server sends back partial results as well as a nextlink to use to retrieve more results. The nextlink is an opaque link which may use $skiptoken to identify the last loaded record.
```

Comment:
>the last loaded record [](start = 235, length = 22)

actually, the skiptoken can be used for storing any state about the request.  Maybe say something like "may use $skiptoken to store state about the request, such as the last read entity."